### ISSUE TITLE
Extract named constants for magic numbers in BackPropagation.ts

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#2219]
## Summary

`src/propagate/BackPropagation.ts` contains numerous hardcoded numeric constants that control training behaviour. These magic numbers make it difficult to understand what each value represents and to tune them consistently.

## Specific magic numbers to extract

In `createBackPropagationConfig()` and surrounding functions:
- **Line ~146**: `10` — maximum generations range for random selection
- **Line ~160**: `10_000` — limit bias scale
- **Line ~162**: `100_000` — limit weight scale
- **Lines ~237-240**: `0.95`, `0.2` — learning rate decay factor, coordination factor
- **Lines ~295-305**: `0.95`, `1.0`, `1.3`, `0.5` — adaptive learning rate thresholds and multipliers

## What needs to be done

1. Define named constants at module scope with clear, descriptive names:
   ```typescript
   const MAX_RANDOM_GENERATIONS = 10;
   const DEFAULT_BIAS_LIMIT_SCALE = 10_000;
   const DEFAULT_WEIGHT_LIMIT_SCALE = 100_000;
   const LEARNING_RATE_DECAY_FACTOR = 0.95;
   const COORDINATION_FACTOR = 0.2;
   const ERROR_IMPROVEMENT_THRESHOLD = 0.95;
   const STAGNATION_BOOST_FACTOR = 1.3;
   ```
2. Replace all magic number usages with the named constants
3. Ensure no behavioural change — this is a pure refactor

## Acceptance criteria

- [ ] All magic numbers in BackPropagation.ts are replaced with descriptively named constants
- [ ] No behavioural change (all existing tests pass unchanged)
- [ ] Constants are defined at module scope with clear names
- [ ] `./quality.sh` passes
- [ ] Australian English used throughout

Part of #2215

**Comment by stservice** [comment]
Claimed by `worker-51361`

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/2226 has been created to address this issue.

**Comment by stservice** [comment]
Automatic recovery: this issue was unassigned from `stservice` because the worker has no active heartbeat for this issue (assigned over 30 minutes ago with no heartbeat recorded). This likely means the worker crashed between claiming the issue and starting work. The issue is now available to be picked up again.

---

Assigned-without-heartbeat recovery (Issue #632)
